File: backend/RecipeApplication/src/main/java/application/services/model/Recipe.java
package application.services.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Duration;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "recipe")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
public class Recipe {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private Duration duration;
    private Integer amountOfPeople;
    private String description;
    private String instructions;
    @OneToMany(cascade = CascadeType.ALL)
    private List<RecipeIngredient> ingredients = new ArrayList<>();
    private String userId;

    @CreationTimestamp
    private LocalDateTime createdAt;
    @UpdateTimestamp
    private LocalDateTime updatedAt;
}

File: frontend/chomp/lib/api/models/recipe.ts
import { DateTime } from 'luxon';

export interface DbRecipe {
  id: number;
  name: string;
  duration: string;
  amountOfPeople: number;
  description?: string;
  instructions: string[];
  user?: DbRecipeUser;
  ingredients: DbIngredient[];
  createdAt?: DateTime;
  updatedAt?: DateTime;
}

export interface DbRecipeUser {
  name: string;
  image: string;
}
